src/main/java/io/xcherry/core/client/Client.java

```diff
@@ -334,4 +337,32 @@ private void publishToLocalQueue(final String processId, final LocalQueueMessage
 
         basicClient.publishToLocalQueue(request);
     }
+
+    /**
+     * Wait for a process execution to complete within the specified timeout seconds.
+     *  1. If the process execution is still running after the timeout seconds, the response will be returned with the timeout field being true.
+     *  2. If the process execution has stopped by system (e.g., due to re-balancing after increasing/decreasing the shards in the server), the response will be returned with the stopBySystem field being true.
```

Comment:
nit: `re-balancing after increasing/decreasing the shards in the server` reblancing could happen without inc/dec shards. maybe just say reblancing is enough